File: C#/1701_AverageWaitingTime.cs
public class Solution {
    public double AverageWaitingTime(int[][] customers) {
        double sum = 0;
        int start = customers[0][0];
        for (int i = 0; i < customers.Length; i++)
        {
            start = Math.Max(start, customers[i][0]);
            start += customers[i][1];
            sum += start - customers[i][0];
        }
        return sum / customers.Length;
    }
}

File: C++/1701_AverageWaitingTime.cpp
class Solution {
public:
    double averageWaitingTime(vector<vector<int>>& customers) {
        double summy = 0;
        int start = customers[0][0];
        for (vector<int> i : customers){
            start = max(start, i[0]);
            start += i[1];
            summy += start - i[0];
        }
        return summy / customers.size();
    }
};